ee/audit/interceptor_ee.go

```diff
@@ -156,19 +167,121 @@ func auditHttp(w *ResponseWriter, r *http.Request) {
 	} else {
 		user = getUser("", false)
 	}
+
 	auditor.Audit(&AuditEvent{
+		Timestamp:   ts,
 		User:        user,
 		Namespace:   x.ExtractNamespaceHTTP(r),
 		ServerHost:  x.WorkerConfig.MyAddr,
 		ClientHost:  r.RemoteAddr,
 		Endpoint:    r.URL.Path,
 		ReqType:     Http,
-		Req:         truncate(string(rb), maxReqLength),
+		Req:         truncate(checkRequestBody(Http, r.URL.Path, string(body)), maxReqLength),
 		Status:      http.StatusText(w.statusCode),
 		QueryParams: r.URL.Query(),
 	})
 }
 
+func maskPasswordFieldsInGQL(req string) string {
+	var gqlReq schema.Request
+	err := json.Unmarshal([]byte(req), &gqlReq)
+	if err != nil {
+		glog.Errorf("unable to marshal gql request %v", err)
+		return req
+	}
+	query, gErr := parser.ParseQuery(&ast.Source{
+		Input: gqlReq.Query,
+	})
+	if gErr != nil {
+		glog.Errorf("unable to parse gql request %+v", err)
+		return req
+	}
+	if len(query.Operations) == 0 {
+		return req
+	}
+	var variableName string
+	for _, op := range query.Operations {
+		if op.Operation != ast.Mutation || len(op.SelectionSet) == 0 {
+			continue
+		}
+
+		for _, ss := range op.SelectionSet {
```

Comment:
You might be able to simplify this a bit
```suggestion
		for _, ss := range op.SelectionSet {
			f, ok := ss.(*ast.Field)
			if !ok {
				continue
			}
			if len(f.Arguments) == 0 {
				continue
			}

			if f.Name == "resetPassword" {
				for _, a := range f.Arguments {
					if a.Name != "input" || a.Value == nil || a.Value.Children == nil {
						continue
					}
					for _, c := range a.Value.Children {
						if c.Name == "password" && c.Value.Kind == ast.Variable {
							variableName = c.Value.String()
						}
					}
				}
			} else if f.Name == "login" {
				for _, a := range f.Arguments {
					if a.Name == "password" && a.Value.Kind == ast.Variable {
						variableName = a.Value.String()
					}
				}
			}
		}
```